sdk/textanalytics/ai-text-analytics/test/utils/recordedClient.ts

```diff
@@ -46,25 +35,33 @@ export const environmentSetup: RecorderEnvironmentSetup = {
   queryParametersToSkip: []
 };
 
-export function createRecordedClient(
-  context: Context,
-  apiKey?: AzureKeyCredential
-): RecordedClient {
-  const recorder = record(context, environmentSetup);
+export type AuthMethod = "APIKey" | "AAD";
 
+export function createClient(
+  authMethod: AuthMethod,
+  options?: TextAnalyticsClientOptions
+): TextAnalyticsClient {
   let credential: AzureKeyCredential | TokenCredential;
-  if (apiKey !== undefined) {
-    credential = apiKey;
-  } else {
-    credential = new ClientSecretCredential(
-      testEnv.AZURE_TENANT_ID,
-      testEnv.AZURE_CLIENT_ID,
-      testEnv.AZURE_CLIENT_SECRET
-    );
+  switch (authMethod) {
+    case "APIKey": {
+      credential = new AzureKeyCredential(env.TEXT_ANALYTICS_API_KEY);
+      break;
+    }
+    case "AAD": {
+      credential = new ClientSecretCredential(
+        env.AZURE_TENANT_ID,
+        env.AZURE_CLIENT_ID,
+        env.AZURE_CLIENT_SECRET
+      );
+      break;
+    }
+    default: {
+      throw Error(`Unsupported authentication method: ${authMethod}`);
+    }
   }
+  return new TextAnalyticsClient(env.ENDPOINT, credential, options);
+}
 
-  return {
-    client: new TextAnalyticsClient(testEnv.ENDPOINT, credential),
-    recorder
-  };
+export function createRecorder(context: Context): Recorder {
```

Comment:
Still needed now that it's just a single call to `record`?

Reply:
I think it is nice to hide `environmentSetup` from the caller context.